swarm/network/stream/delivery.go

```diff
@@ -244,22 +153,16 @@ func (d *Delivery) handleChunkDeliveryMsg(ctx context.Context, sp *Peer, req int
 	case *ChunkDeliveryMsgSyncing:
 		msg = (*ChunkDeliveryMsg)(r)
 		mode = chunk.ModePutSync
+	case *ChunkDeliveryMsg:
```

Comment:
One of the tests is actually sending a `ChunkDeliveryMsg` and not the other two specific types, therefore failing.

This is not code that is hit in production, but I decided to add this here temporarily, rather than dig into the test.

It is not clear why this test was not failing with the previous code.